src/pages/visualization-view/connector.js

```diff
@@ -13,14 +13,16 @@ const mapStateToProps = state => {
         isSaving: visualizationSaving(state),
         isSaveFailure: visualizationSaveFailure(state),
         isSaveSuccess: visualizationSaveSuccess(state),
-        isSaveable: isVisualizationSaveable(state)
+        isSaveable: isVisualizationSaveable(state),
+        updateQuery: visualizationID(state) !== undefined
```

Comment:
maybe rename to `shouldUpdateQuery` or `isQueryUpdateable` - Ben

Reply:
On top of that, combine in `updateQuery || props.isSaving` in here too - Ben and Tim

Reply:
Refactored to just not use updateQuery.